pkgs/by-name/dr/draupnir/package.nix

```diff
@@ -0,0 +1,100 @@
+{ lib
+, fetchFromGitHub
+, makeWrapper
+, nodejs
+, matrix-sdk-crypto-nodejs
+, sqlite
+, mkYarnPackage
+, fetchYarnDeps
+, nixosTests
+}:
+
+# docs: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md#yarn2nix-javascript-yarn2nix
+let
+  hashesFile = builtins.fromJSON (builtins.readFile ./hashes.json);
+in
+  mkYarnPackage rec {
+    pname = "draupnir";
+    version = "2.0.0-beta.6";
+    src = fetchFromGitHub {
+      owner = "the-draupnir-project";
+      repo = "Draupnir";
+      rev = "v${version}";
+      hash = "sha256-s1LWXVwY+7LD7cJtZW7mBLsdpB499zS/nDsJ7qaQDfg=";
+    };
+
+    nativeBuildInputs = [
+      makeWrapper
+      sqlite
+    ];
+
+    offlineCache = fetchYarnDeps {
+      name = "${pname}-yarn-offline-cache";
+      yarnLock = src + "/yarn.lock";
+      hash = hashesFile.yarn_offline_cache_hash;
+    };
+
+    yarnLock = src + "/yarn.lock";
+    packageJSON = ./package.json;
+
+    #prebuild phase
+    preBuild = ''
+      # copy built modules to package...
+      echo "Copying built matrix-sdk-crypto-nodejs modules to package..."
+      cp -a ${matrix-sdk-crypto-nodejs}/lib/node_modules/* node_modules/
+      echo "Adding version.txt..."
+      mkdir -p deps/draupnir/
+      echo "${version}-nix" > deps/draupnir/version.txt
+
+      sed -i 's/corepack //g' deps/draupnir/package.json
```

Comment:
(Copy paste from commit description)
fixup! draupnir: init at 2.0.0-beta.6 - ugly sed hack to drop corepack

Fixes build with corepack upstream introduced by https://github.com/the-draupnir-project/Draupnir/pull/472
Upstream may move to yarn v4, need to investigate options. (https://github.com/the-draupnir-project/Draupnir/issues/475)
Looks like the recommended option is to use a fixed-output derivation?
 
Sources:
- https://www.reddit.com/r/NixOS/comments/1f8tq94/its_2024_has_anyone_figured_out_how_to_build_a/
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/cluster/tilt/assets.nix

Any suggestions? I'm sure this is an ugly, unnecessary hack, but I couldn't find any documentation on the topic.